internal/source/dispatcher.go

```diff
@@ -96,13 +99,18 @@ func (d *Dispatcher) Dispatch(dispatch PluginDispatch) error {
 		return fmt.Errorf("while getting source client for %s: %w", dispatch.pluginName, err)
 	}
 
+	kubeconfig, err := plugin.GenerateKubeConfig(d.restCfg, dispatch.pluginContext.RBAC)
+	if err != nil {
+		return fmt.Errorf("while generating kube config for %s: %w", dispatch.pluginName, err)
+	}
+
 	ctx := dispatch.ctx
 	out, err := sourceClient.Stream(ctx, source.StreamInput{
 		Configs: dispatch.pluginConfigs,
 		Context: source.StreamInputContext{
 			IsInteractivitySupported: dispatch.isInteractivitySupported,
 			ClusterName:              dispatch.cfg.Settings.ClusterName,
-			KubeConfig:               dispatch.cfg.Settings.Kubeconfig,
```

Comment:
This might break existing functionality. We might as well remove kube config from helm

Reply:
2 cents from my side: I like that option however, even Argo Workflow which is quite popular project, it doesn't support external/multi-cluster (they have issue opened for more that 2years now). If someone wants to have such functionality, they can always use https://github.com/virtual-kubelet/virtual-kubelet#providers. Of course, it brings additional maintainer cost, but it's also a more powerful and “safer” option.

Maybe at the end it just brings not necessary complexity to the Botkube itself? Probably it's a question to @brampling whether we should leave that functionality or not.

---

However, if we will decide to leave that functionality, it would be good to name it like `context.remote.kubeconfig`? so it is easier to distinguish current cluster RBAC usage vs external one.